backend/src/test/java/com/github/hamideh6182/controller/BookControllerTest.java

```diff
@@ -248,4 +249,40 @@ void decreaseBookQuantityTest_WhenQuantityMinusOne() throws Exception {
                         }
                                                 """));
     }
+
+    @Test
+    @DirtiesContext
+    @WithMockUser("user2")
+    void checkoutBook_withValidInput_shouldReturnNewBook() throws Exception {
+        mongoUserRepository.save(mongoUser2);
```

Comment:
Maybe you should name the test with when&then. For example: whenCheckoutBook_then_ReturnNewBook.